test/wrappers/ortools_test.rb

```diff
@@ -5368,4 +5368,28 @@ def test_relations_sent_to_ortools_when_different_lapses
       end
     }
   end
+
+  def test_always_one_time_window_provided_for_rests
+    problem = VRP.basic
+    problem[:rests] = [{ id: 'rest', duration: 10 }]
+    problem[:vehicles][0][:rest_ids] = ['rest']
+
+    [[], [{ start: 2, end: 3 }]].each{ |rest_tws|
+      OptimizerWrapper.config[:services][:ortools].stub(
+        :run_ortools,
+        lambda { |pb, _, _|
+          assert_equal 1, pb.vehicles.first.rests.first.time_windows.size
```

Comment:
I suspect that there is a bug in the new TW collection logic. 

Because for the first `[]` case, the vehicle should have no rests since the `simplify_vehicle_pause` simplification is active (that's what I see when I test on local with this instance (without the modification I suggested above), the pause is really eliminated). So I don't quite understand how this test passes, any idea?

Reply:
Oh I just found it, I think, I called `wrapper_vrp` and this test calls `solve` so it skips the eliminations. However, because of that we basically test with a state that cannot happen in reality